### PR TITLE
Implement ZStream#takeRight

### PR DESCRIPTION
**Pull request by paulpdaniels**
Relates to #4886

### CODE REVIEW

File: streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala [hunk 1]
```diff
@@ -1980,7 +1980,23 @@ class ZStream[-R, +E, +A](val channel: ZChannel[R, Any, Any, Any, E, Chunk[A], A
...
-    ???
+    if (n <= 0) ZStream.empty
+    else
+      new ZStream(
+        ZChannel.unwrap(
+          for {
+            queue <- Queue.sliding[A](n)
+          } yield {
+            lazy val reader: ZChannel[Any, E, Chunk[A], Any, E, Chunk[A], Unit] = ZChannel.readWith(
+              (in: Chunk[A]) => ZChannel.fromEffect(queue.offerAll(in)) *> reader,
```
Comment (iravid) [inline streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala:1991]: Would you be alright with using a plain mutable Queue here? should be much faster.

Comment (paulpdaniels) [reply streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala:1991]: Would @adamgfraser 's RingBuffer work here as well?

Comment (adamgfraser) [reply streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala]: We need a `SingleThreadedRingBuffer` here. Let me do a quick PR to expose the functionality we want.

Comment (adamgfraser) [reply streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala]: Implemented in #4951.

Comment (adamgfraser) [reply streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala]: @paulpdaniels This is available on the `series/2.x` branch now.

**Comment by paulpdaniels** [comment]
@iravid @adamgfraser comments addressed. Not sure what is going on with CI seems unrelated to the changes

**Comment by adamgfraser** [comment]
@paulpdaniels Can you merge upstream changes?

**Comment by adamgfraser** [comment]
You may also need to run `sbt prepare`.

**Review by iravid — APPROVED** [review]
This is awesome! Ok we just need an `sbt prepare` here as well and we can merge.

**Review by iravid — APPROVED** [review]

**Pull request merged**